data/crac/crac-impl/src/main/java/com/powsybl/openrao/data/cracimpl/RaUsageLimitsAdderImpl.java

```diff
@@ -7,31 +7,27 @@
 
 package com.powsybl.openrao.data.cracimpl;
 
+import com.powsybl.openrao.commons.OpenRaoException;
 import com.powsybl.openrao.data.cracapi.*;
 
 import java.util.*;
 
-import static com.powsybl.openrao.commons.logs.OpenRaoLoggerProvider.BUSINESS_WARNS;
-import static com.powsybl.openrao.data.cracimpl.AdderUtils.assertAttributeNotNull;
-
 /**
  * @author Martin Belthle {@literal <martin.belthle at rte-france.com>}
  */
 public class RaUsageLimitsAdderImpl implements RaUsageLimitsAdder {
-    public static final String RA_USAGE_LIMITS = "RaUsageLimits";
     CracImpl owner;
-    private Instant instant = null;
+    private final Instant instant;
     private final RaUsageLimits raUsageLimits = new RaUsageLimits();
 
     RaUsageLimitsAdderImpl(CracImpl owner, String instantName) {
         Objects.requireNonNull(owner);
         this.owner = owner;
         List<Instant> instants = this.owner.getSortedInstants().stream().filter(cracInstant -> cracInstant.getId().equals(instantName)).toList();
         if (instants.isEmpty()) {
-            BUSINESS_WARNS.warn("The instant {} registered in the crac creation parameters does not exist in the crac. Its remedial action limitations will be ignored.", instantName);
-        } else {
-            this.instant = instants.get(0);
+            throw new OpenRaoException(String.format("The instant %s registered in the crac creation parameters does not exist in the crac. Its remedial action limitations will be ignored.", instantName));
```

Comment:
you dont't know that the user is using a crac creation parameters, maybe they are calling the adder API. plus, the RA limitations will not only be ignored, since you threw an exception, so the user's app will crash.
```suggestion
            throw new OpenRaoException(String.format("The instant %s does not exist in the crac.", instantName));
```

Reply:
Indeed, i committed to quickly.